### PR TITLE
CrystalLintBear.py: Add `ameba`

### PR DESCRIPTION
**Pull request by sangamcse**
This adds a local lint bear for `Crystal` language
wrapping `ameba`.

Closes https://github.com/coala/coala-bears/issues/2332

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by sangamcse** [comment]
@jayvdb, need help in setting up ameba for the Travis.

File: shard.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: coala-bears
+version: 0.12.0.dev99999999999999
```
Comment (sangamcse) [inline shard.yml]: @jayvdb, I am a bit confused here about the name, version, authors, etc. This `version` part is mandatory according to the shard, BTW.

File: shard.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+authors:
+  - The coala developers <coala-devel@googlegroups.com>
```
Comment (jayvdb) [inline shard.yml]: This file needs to be generated from a moban template like the others are, using Requirement classes.

File: .ci/deps.sh [hunk 3]
```diff
@@ -58,3 +58,10 @@ if [ ! -e ~/.local/bin/flawfinder ]; then
   sed -i '1s/.*/#!\/usr\/bin\/env python2/' ~/.local/bin/flawfinder
   chmod +x ~/.local/bin/flawfinder
 fi
+
+# Shards commands
+if [ ! -e ~/.local/bin/ameba ]; then
+  shards install
```
Comment (jayvdb) [inline .ci/deps.sh]: Need to add a `ShardsRequirement` to package_manager.

But .. as you are using travis apt , and circle is disabled anyway, another option is to enhance DistributionRequirement to support custom repis.

File: shard.yml (hunk 1 above)
Comment (jayvdb) [reply shard.yml]: I think you have the right content for those. But they need to come from a template

File: .ci/deps.sh (hunk 3 above)
Comment (sangamcse) [reply .ci/deps.sh]: Enhancing `DistributionRequirement` is a better option as this is required for only one bear right now and I don't think `shards` will be required for any other bear in future also.

Comment (sangamcse) [reply .ci/deps.sh]: @jayvdb, `Crystal` and `ameba` both can be installed using `brew`. 
Ref [crystal-lang using brew](https://crystal-lang.org/docs/installation/on_mac_osx_using_homebrew.html), [ameba using brew](https://github.com/veelenga/ameba#os-x).

Note: I have also written brew dependency in the bear.

File: bears/crystal/CrystalLintBear.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+from coalib.bearlib.abstractions.Linter import linter
+from dependency_management.requirements.ShardRequirement import (
+    ShardRequirement)
+
+
+@linter(executable='ameba',
+        output_format='regex',
+        output_regex=r'.+:(?P<line>\d+):(?P<column>\d+)\n'
+                     r'.+31m(?P<origin>[A-Za-z]+): (?P<message>.*)')
```
Comment (jayvdb) [inline bears/crystal/CrystalLintBear.py]: 31m is ansi stuff?

we have a cleaner for that : `strip_ansi`

File: bears/crystal/CrystalLintBear.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
+from coalib.bearlib.abstractions.Linter import linter
+from dependency_management.requirements.ShardRequirement import (
```
Comment (jayvdb) [inline bears/crystal/CrystalLintBear.py]: blocked by https://gitlab.com/coala/package_manager/issues/200

File: bears/crystal/CrystalLintBear.py (hunk 4 above)
Comment (sangamcse) [reply bears/crystal/CrystalLintBear.py]: Yes.
> 31m is ansi stuff?

Okay, I will look into it.
> we have a cleaner for that : strip_ansi

**Comment by jayvdb** [comment]
This is blocked - please stop pushing.

**Pull request closed without merging**